web/src/components/deployments-detail-page/pipeline/index.tsx

```diff
@@ -54,16 +54,27 @@ const findDefaultActiveStage = (
   return stages[stages.length - 1];
 };
 
+const isStartedStage = (stage: Stage): boolean => {
+  return stage.status !== StageStatus.STAGE_NOT_STARTED_YET;
+};
+
 const createStagesForRendering = (
   deployment: Deployment.AsObject | undefined
 ): Stage[][] => {
   if (!deployment) {
     return [];
   }
 
-  const stages: Stage[][] = [];
-  const visibleStages = deployment.stagesList.filter((stage) => stage.visible);
+  let visibleStages: Stage[] = [];
+  if (deployment.deployTargetsByPluginMap?.length === 0) {
```

Comment:
I'm not sure about the optional chaining feature in TypeScript, but I think that the `deployment.deployTargetsByPluginMap?.length` will be `undefined` when the `deployment.deployTargetsByPluginMap` is null/undefined. And `undefined === 0` will be `false`.

Reply:
Got your point 👀 Should I change this to be as same as this one https://github.com/pipe-cd/pipecd/blob/master/web/src/components/deployments-detail-page/deployment-detail/index.tsx#L80
cc @hongky-1994

Reply:
Thanks both, addressed by https://github.com/pipe-cd/pipecd/pull/5950/commits/1460183ce018092d177f69fbacf326c4bbaee51b 🙏